src/promptflow/promptflow/_sdk/_serving/flow_invoker.py

```diff
@@ -202,3 +199,27 @@ def _dump_invoke_result(self, invoke_result):
                 invoke_result.output, base_dir=self._dump_to, sub_dir=Path(".promptflow/output")
             )
             dump_flow_result(flow_folder=self._dump_to, flow_result=invoke_result, prefix=self._dump_file_prefix)
+
+
+class AsyncFlowInvoker(FlowInvoker):
+    async def _invoke_async(self, data: dict, run_id=None, disable_input_output_logging=False):
+        self._invoke_context(data, disable_input_output_logging)
+        return await self.executor.exec_line_async(data, run_id=run_id, allow_generator_output=self.streaming())
```

Comment:
@thy09 I removed the index=0 paramter and remove the following code.

if isinstance(result.output, dict) and LINE_NUMBER_KEY in result.output:
            # Remove line number from output
            del result.output[LINE_NUMBER_KEY]